applications/conceptual-model-editor/src/app/diagram/reactflow/components/drag-edge-canvas-menu.tsx

```diff
@@ -0,0 +1,52 @@
+import { XYPosition } from "reactflow";
+import { useMenuOptions } from "./menu-options";
+import { useState } from "react";
+
+export const useCanvasMenuOptions = () => {
+
+  const { 
+    MenuOptionsGeneral,
+    isMenuOptionsOpen,
+    openMenuOptions: open,
+    closeMenuOptions: close, 
+    } = useMenuOptions();
+
+    const [canvasMenuXYPosition, setCanvasMenuXYPosition] = useState<XYPosition>();
+
+    const CanvasMenuOptions = (props: {    
+            openCreateClassDialogHandler: () => void;  
+            openSelectClassDialog: () => void;
+        }) => {
+
+
+        const handlersWithTexts: Array<[(() => void) | undefined, string]> = [      
```

Comment:
I get this is based on the API we already have. But it would be better to make it more explicit. 
Instead of using Array or pairs .. we can use array of objects where each object can have callback and label.